File: android/app/src/test/java/miko/todayinspacehistory/helpers/StringHTMLDecodingTests.kt
package miko.todayinspacehistory.helpers

import miko.todayinspacehistory.util.decodedHTMLEntities
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class StringHTMLDecodingTests {
    @Test
    fun decodesNamedEntity() {
        assertEquals("&", "&amp;".decodedHTMLEntities)
    }

    @Test
    fun decodesNumericEntity() {
        // &#0146; is the Windows-1252-style curly apostrophe; HTML decode yields U+2019.
        assertEquals("\u2019", "&#0146;".decodedHTMLEntities)
    }

    @Test
    fun plainTextUnchangedAsideFromTrimming() {
        assertEquals("hello", "hello".decodedHTMLEntities)
    }
}

File: ios/TodayInSpaceHistoryTests/Helpers/StringHTMLDecodingTests.swift
//
//  StringHTMLDecodingTests.swift
//  TodayInSpaceHistoryTests
//

import XCTest
@testable import TodayInSpaceHistory

final class StringHTMLDecodingTests: XCTestCase {
    func testDecodesNamedEntity() {
        XCTAssertEqual("&amp;".decodedHTMLEntities, "&")
    }

    func testDecodesNumericEntity() {
        // &#0146; is the Windows-1252-style curly apostrophe; HTML decode yields U+2019.
        XCTAssertEqual("&#0146;".decodedHTMLEntities, "\u{2019}")
    }

    func testPlainTextUnchangedAsideFromTrimming() {
        XCTAssertEqual("hello".decodedHTMLEntities, "hello")
    }
}
